### ISSUE TITLE
`terraform import` fails for resources using `for_each`

### ISSUE THREAD

**Issue by erlendr** [hashicorp/terraform#32329]
### Terraform Version

```shell
v1.3.0
v1.3.1
v1.3.2
v1.3.3
v1.3.4
v1.3.5
v1.3.6
v1.4.0-alpha20221109
```

### Terraform Configuration Files

```terraform
terraform {
  required_providers {
    random = {
      source  = "hashicorp/random"
      version = "3.4.3"
    }
  }
}

locals {
  ids = ["id1"]
}

resource "random_id" "test" {
  for_each    = toset(local.ids)
  byte_length = 4
}
```

### Debug Output

https://gist.github.com/erlendr/7b568080783a7ce1f61b37a7be60f4eb

### Expected Behavior

New resource instance `id2` imported into state.

### Actual Behavior

Command outputs "Import successful!", but new instance was not imported. `terraform.tfstate` has no changes.

### Steps to Reproduce

1. `terraform init`
2. `terraform apply --auto-approve`
3. `terraform import 'random_id.test["id2"]' p-9hUg`

Example: https://github.com/erlendr/tf-foreach-import-bug

### Additional Context

Works as expected in `v1.2.9`. I suspect https://github.com/hashicorp/terraform/issues/31283 may be the root cause.

### References

- #31283

**Comment by devbeard** [comment]
I reproduced this on Windows in Powershell. 

Outcome: 
The `id2` did not appear in the state file. 

Release/version used: 
`terraform_1.3.5_windows_amd64.zip`

Powershell-edition of the import command: 
`terraform import 'random_id.test[\"id2\"]' 'p-9hUg'`

**Comment by jbardin** [comment]
Hi @erlendr,

Thanks for filing the issue! This seems to be a bug from caused by v1.3 attempting to keep some backward compatibility with the old import mechanism, while also using a more details plan. The problem here is that in order to import an instance, there must be a configuration for that resource instance, i.e. running a `plan` before `import` would show that the about-to-be imported instance is going to be created. If you add `"id2"` to your `ids` set, the import works correctly.

The ability to always import into a resource using `for_each` is still not complete, as shown in issues like #28188. The next step for `import` will likely involve the ability to declare and plan the imports via configuration, which can allow for dealing with multiple instances and better review of the process as a whole.

Thanks!

**Comment by devbeard** [comment]
@jbardin Just for my own understanding since you called it a "bug caused by v1.3" but removed the bug label, is this a bug or intended behavior?

**Comment by jbardin** [comment]
Sorry, I meant to leave `bug` and remove `new`. It's a bug that it completed with no feedback about the missing import, but is otherwise the intended behavior that the resource instance must be included in the configuration.

**Comment by MB175** [comment]
I started debugging this today, the provider SDK does work til the first "GRPC - Configure" command, and than it silently crashes.

**Comment by geekofalltrades** [comment]
I just upgraded from 1.2.9 to 1.4.6 and am facing a similar issue with a resource made conditional using `count`.

```hcl
resource "resource_type" "my_resource"
  count = condition ? 1 : 0
}
```

I had some code that was importing this with `terraform import resource_type.my_resource the_resource` (no count index), and it was working in 1.2.9. Now it's failing, but not with an error. Terraform is printing "Import successful!", but not actually importing anything, as reported, and more importantly, is not exiting with an error, so it took me a while to figure out where things were going wrong. Doing `terraform import resource_type.my_resource[0] the_resource` instead works as expected, and is more correct, but it's not helpful that Terraform was failing silently in the previous case instead of with an error.

**Comment by morremeyer** [comment]
Can confirm that this happens on 1.4.6, we're affected by it, too. Is there an ETA for a fix, and would it help to open a support ticket as a business customer?

**Comment by morremeyer** [comment]
Adding a minimal reproduction here to verify for our case:

First, export your GitHub PAT:

```sh
export GITHUB_TOKEN="YOUR_TOKEN"
```

```hcl
terraform {
  required_providers {
    github = {
      source  = "integrations/github"
      version = "5.28.1"
    }
  }
}

provider "github" {
  owner = "morremeyer"
}

resource "github_repository" "test_repo" {
  for_each = toset(["terraform-issue-32329-reproduction"])
  name        = each.key
}
```

**Comment by b2sc** [comment]
I think this is still a bug as [jbardin](https://github.com/jbardin) said for not warning about skipping the import. It would have helped me if this would have been improved/ fixed. Thanks!

**Comment by asoc** [comment]
After referring to the docs https://developer.hashicorp.com/terraform/language/import

can confirm still an issue in 1.5.5:
```
❯ terraform -version
Terraform v1.5.5
on darwin_arm64
```

Plan output without `imports`:
 ```
      # module.repositories.aws_ecr_repository.ecr["seed/api"] will be created
      + resource "aws_ecr_repository" "ecr" {
          + arn                  = (known after apply)
          + force_delete         = true
          + id                   = (known after apply)
          + image_tag_mutability = "MUTABLE"
          + name                 = "seed/api"
...
```

after adding this:
```
import {
  to = aws_ecr_repository.ecr["seed/api"]
  id = "seed/api"
}
```

I get this:

```
    Error: Config generation for count and for_each resources not supported.

    Your configuration contains an import block with a "to" address of aws_ecr_repository.ecr["seed/api"]. This resource instance does not exist in configuration.

    If you intended to target a resource that exists in configuration, please double-check the address. Otherwise, please remove this import block or re-run the plan without the -generate-config-out flag to ignore the import block.```

**Comment by jbardin** [comment]
Closed as working in v1.7
